src/compiler/diagnosticMessages.json

```diff
@@ -3365,6 +3365,10 @@
         "category": "Error",
         "code": 2809
     },
+    "Expected {0} arguments, but got {1}. TypeScript may need a JSDoc hint that the call to 'new Promise()' produces a 'Promise<void>'": {

```

Comment:
I believe this was the message [suggested in the issue](https://github.com/microsoft/TypeScript/issues/46570#issuecomment-955106669), so it make sense to use it, but this wording strikes me as off as I'm not sure that we have any error messages that effectively "blame" TypeScript itself for needing a hint about the code's behavior, at least not in this particular wording. I would have written something more like

> This promise may need a JSDoc-style type that the call to 'new Promise()' produces a 'Promise<void>'

To shift things around to pointing at the particular code that's at fault, not at the language itself.

But, this is bikeshedding. @DanielRosenwasser @andrewbranch

Reply:
It adds a type tag: `/** @type {Promise<void>} */(new Promise(resolve => resolve()));`

Reply:
How about

```
'new Promise()' needs a JSDoc hint to produce a 'resolve' that can be called without arguments.
```

- `'resolve'` could be substituted with an interpolation to use the parameter name the user wrote.
- The first interpolation (`{0} arguments`) is unnecessary as we know it’s `0`.

(cc @DanielRosenwasser)

Reply:
Nice, this provides better feedback indeed :)